hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiBoundaryTypeAttributeEnricher.java

```diff
@@ -144,11 +144,18 @@ private void enrichHostHeader(Event event) {
     }
   }
 
-  private Optional<String> getGrpcAuthority(Event event) {
+  private Optional<String> getGrpcHostHeader(Event event) {
```

Comment:
@kotharironak should I add test over here also?

Reply:
Yes, add a test with an event not having authority but required tags like rpc.sytem and rec.request.metadata.host, and hostHeader fall back to metadata.host value.

Reply:
Refer the test file `ApiBoundaryTypeAttributeEnricherTest`, test cases `testEnrichEventWithGrpcNoAuthority` and similar.

Reply:
resolved

Reply:
There is a test `testEnrichEventWithGrpcAuthority` in the same file, can you also add the below attribute to it.

```
addAttributeToEvent(
        innerEntrySpan,
        RPC_REQUEST_METADATA_HOST.getValue(),
        AttributeValue.newBuilder().setValue("testHost2").build());
```

Reply:
done